python/ray/tune/utils/resource_updater.py

```diff
@@ -0,0 +1,127 @@
+from typing import Optional, Dict, Any
+import logging
+import os
+import time
+
+import ray
+from ray import ray_constants
+from ray._private.resource_spec import NODE_ID_PREFIX
+from ray.tune.resources import Resources
+
+logger = logging.getLogger(__name__)
+
+TUNE_STATE_REFRESH_PERIOD = 10  # Refresh resources every 10 s
+
+
+def _to_gb(n_bytes):
+    return round(n_bytes / (1024 ** 3), 2)
+
+
+class ResourceUpdater:
+    """Periodic Resource updater for Tune."""
```

Comment:
I think we should document here the class is operating under the assumption that it stays within one cycle of ray.init(). Otherwise, `_last_resource_refresh  > 0` cannot really tell us about whether init is done.

Reply:
hmmm, what does `stay in one cycle` mean? The semantics is 1. update when get resources, or when `update_avail_resources()` is called. 2. immediate update if not updated before and ray is initialized 3. after first update, the interval between update is at least `refresh_period`

Reply:
yes, I mean to be used with 
ray.init()
updater = ResourceUpdater()
# its life cycle
ray.shutdown()
# not meaningful to use updater again.

I don't think it's meant to be used with multiple init(), shutdown() and then init()

Reply:
I clarify it in the doc